Explorer/Assets/Scripts/Global/Dynamic/MainSceneLoader.cs

```diff
@@ -330,6 +331,25 @@ private async UniTask ChangeRealmAsync(CancellationToken ct)
             await realmController.SetRealmAsync(URLDomain.FromString(startingRealm), ct);
         }
 
+        private async UniTask InitializeFeatureFlagsAsync(IWebRequestController webRequestController, CancellationToken ct)
```

Comment:
I believe in this method there are some encapsulation/abstraction problems:

- `IFeatureFlagsProvider` is not used but `HttpFeatureFlagsProvider` is instantiated directly. `IFeatureFlagsProvider` should be passed as a dependency instead, probably created in `Container`
- You should not interfere with internals of `FeatureFlagsCache` as, thus, via the open setter you violate encapsulation. I propose you to pass `DefaultFeatureFlagsCache` (the final implementation with the open setter) to `HttpFeatureFlagsProvider` from the container and `HttpFeatureFlagsProvider` will set the configuration. You will remove the open setter from the interface this way. In that case it's also not entirely clear why you need an interface. What have you tried to abstract with it? I don't see any application apart from the open setter there.

Reply:
Made the change here: https://github.com/decentraland/unity-explorer/pull/1279/commits/35edb72453721ee65c2cf02edaf25b5766a9fdb8
I do find a problem. If you want to fetch the feature flags from another url or app name by any reason through `HttpFeatureFlagsProvider` it will always replace the current FFs in the cache, which might not be desired, being less versatile. We dont have that use case yet tho, so i guess it wont be a problem. WDYT?

Reply:
If this case is real I can propose the similar scheme as with `PlacesAPIService` and `IPlacesAPIClient`: 
- the former has all the caching capabilities
- the latter has requests capabilities only

In this case you will be able to communicate with the "raw" client if such necessity arises but the proper layering remains for the customers of the higher-level API